system/HTTP/OutgoingRequest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HTTP;
+
+/**
+ * Representation of an outgoing, client-side request.
+ */
+class OutgoingRequest extends Message implements OutgoingRequestInterface
```

Comment:
Why doesn't this extend Request?

Reply:
Okay reading the rest of the code I see I must have misunderstood your UML diagram: it goes Message > OutgoingRequest > Request.

This seems very unintuitive; why don't we have a generic Request with common code that Incoming and Outgoing both extend from?

Reply:
Answered in main thread!